lnp2p/src/bolt/bolt2.rs

```diff
@@ -151,28 +180,12 @@ impl lightning_encoding::LightningDecode for ChannelType {
     fn lightning_decode<D: io::Read>(
         d: D,
     ) -> Result<Self, lightning_encoding::Error> {
-        let mut flags = FlagVec::lightning_decode(d)?;
-        if flags.shrink() {
-            return Err(lightning_encoding::Error::DataIntegrityError(s!(
-                "non-minimal channel type encoding"
-            )));
-        } else if flags.as_inner() == &[] as &[u8] {
-            return Ok(ChannelType::Basic);
-        }
-
-        let mut iter = flags.iter();
-        match (iter.next(), iter.next(), iter.next()) {
-            (Some(12), None, None) => Ok(ChannelType::StaticRemotekey),
-            (Some(12), Some(20), None) => {
-                Ok(ChannelType::AnchorOutputsStaticRemotekey)
-            }
-            (Some(12), Some(22), None) => {
-                Ok(ChannelType::AnchorsZeroFeeHtlcTxStaticRemotekey)
-            }
-            _ => Err(lightning_encoding::Error::DataIntegrityError(s!(
-                "invalid combination of channel type flags"
-            ))),
-        }
+        let flags = FlagVec::lightning_decode(d)?;
+        ChannelType::try_from(flags)
+    }
+    fn lightning_deserialize(data: impl AsRef<[u8]>) -> Result<Self, Error> {
+        let flags = FlagVec::lightning_deserialize(data)?;
```

Comment:
This method still needs to be completed.

To clarify the problem, I wrote the test bellow:

```lang=rust
    #[test]
    fn decode_static_remotekey_channel_type() {
        // Byte array expected by Flagvec::lightning_deserialize
        let bytes = [0u8, 2u8, 0u8, 16u8];
        // That's Working!
        let msg = ChannelType::lightning_deserialize(&bytes);
        assert_eq!(true, msg.is_ok());


        // The byte array received by clightning
        let bytes = [16u8, 0u8];
        // Fail!
        let msg = ChannelType::lightning_deserialize(&bytes);
        assert_eq!(true, msg.is_ok())
    }
```

As tested above, we need to convert the information to little endian and add two more bytes to the left containing information about the size of the original array.